app/src/main/res/layout/activity_main.xml

```diff
@@ -9,68 +9,68 @@
+            android:layout_marginTop="5dp"
+            android:text="Next"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@id/buttonRefresh"
+            app:layout_constraintTop_toBottomOf="@id/imageView" />
 
-        <Button
+        <androidx.appcompat.widget.AppCompatButton
             android:id="@+id/buttonRefresh"
             android:layout_width="120dp"
             android:layout_height="50dp"
             android:layout_marginTop="5dp"
             android:text="Refresh"
             app:layout_constraintEnd_toStartOf="@id/buttonNext"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/imageView"/>
+            app:layout_constraintTop_toBottomOf="@id/imageView" />
 
-        <Button
-            android:id="@+id/buttonNext"
-            android:layout_width="120dp"
-            android:layout_height="50dp"
-            android:layout_marginTop="5dp"
-            android:text="Next"
+        <androidx.appcompat.widget.AppCompatImageView
+            android:id="@+id/imageView"
+            android:layout_width="300dp"
+            android:layout_height="300dp"
+            android:layout_marginTop="20dp"
             app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toEndOf="@id/buttonRefresh"
-            app:layout_constraintTop_toBottomOf="@id/imageView"/>
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/textView" />
 
-        <TextView
+        <androidx.appcompat.widget.AppCompatTextView
             style="@style/Widget.Template.TextView.Clickable"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/text_normal"
             android:gravity="center"
-            android:layout_marginTop="@dimen/dimen_normal_display_text"
             android:padding="15dp"
             android:text="Clickable text"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/buttonRefresh"
+            app:layout_constraintTop_toBottomOf="@id/buttonRefresh" />
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/textView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            android:gravity="center"
+            android:text="Hello World!"
             app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintBottom_toBottomOf="parent"/>
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
 
     </androidx.constraintlayout.widget.ConstraintLayout>
```

Comment:
Can I assume you applied auto-formatting here?

If so, you can see that the the view components are ordered alphabetically, we don't want that. We want view components to be ordered from **top to down** (The view components shown at the top, should be placed at the top of the file as well)

Please confirm if this was the case, as I remember I used to have this problem as well before 🤔

Reply:
Yes, I use auto-formatting here.